File: ios/FOSinOC/FOSinOC/help/CustomizedSegueLeftToRight.m
//
//  CustomizedSegueLeftToRight.m
//  FOSinOC
//
//  Created by MoonSlides on 16/4/14.
//  Copyright © 2016年 李龑. All rights reserved.
//

#import "CustomizedSegueLeftToRight.h"

@implementation CustomizedSegueLeftToRight

-(void)perform{
    UIView *sourceView = self.sourceViewController.view;
    UIView *destinationView = self.destinationViewController.view;
    
    CGFloat screenWidth = [UIScreen mainScreen].bounds.size.width;
    CGFloat screenHeight = [UIScreen mainScreen].bounds.size.height;
    
    destinationView.frame = CGRectMake(sourceView.frame.origin.x + screenWidth, sourceView.frame.origin.y, screenWidth, screenHeight);
    UIWindow *window = [UIApplication sharedApplication].keyWindow;
    [window insertSubview:destinationView aboveSubview:sourceView];
    
//    CGRect destinationFrame = destinationView.frame;
//    
//    destinationFrame = CGRectMake(sourceView.frame.origin.x, sourceView.frame.origin.y, destinationFrame.size.width, destinationFrame.size.height);
    CGRect destinationFrame = CGRectMake(sourceView.frame.origin.x, sourceView.frame.origin.y, screenWidth, screenHeight);
    
    
    [UIView animateWithDuration:0.35 animations:^void{
    
        destinationView.frame = destinationFrame;
   
    }completion:^(BOOL finished){
        
        [self.sourceViewController presentViewController:self.destinationViewController animated:false completion:nil];
    }];
}

@end

File: ios/FoodOrderingSystem/FoodOrderingSystem/Help/CustomizedSegueLeftToRight.swift
//
//  CustomizedSegueLeftToRight.swift
//  FoodOrderingSystem
//
//  Created by MoonSlides on 16/2/4.
//  Copyright © 2016年 李龑. All rights reserved.
//

import UIKit

class CustomizedSegueLeftToRight: UIStoryboardSegue {

    override func perform() {
        
        let sourceViewController = self.sourceViewController.view as UIView
        let destinationViewController = self.destinationViewController.view as UIView
        
        let screenWidth = UIScreen.mainScreen().bounds.size.width
        let screenHeight = UIScreen.mainScreen().bounds.size.height
        
        destinationViewController.frame = CGRectMake(sourceViewController.frame.origin.x + screenWidth, sourceViewController.frame.origin.y, screenWidth, screenHeight)
        
        let window = UIApplication.sharedApplication().keyWindow
        window?.insertSubview(destinationViewController, aboveSubview: sourceViewController)
        
        var destinationFrame = destinationViewController.frame
        destinationFrame = CGRectMake(sourceViewController.frame.origin.x, sourceViewController.frame.origin.y, destinationFrame.size.width, destinationFrame.size.height)
        
        UIView.animateWithDuration(0.35, animations: {
            () -> Void in
            //sourceViewController.frame = CGRectOffset(sourceViewController.frame, -sourceViewController.frame.size.width, 0)
                destinationViewController.frame = destinationFrame
            
            }) {(Finished) -> Void in
                self.sourceViewController.presentViewController(self.destinationViewController as UIViewController,
                    animated: false,
                    completion: nil)
                
        }

    }

}
